### DISCUSSION TITLE
Unit Testing a created pipeline

### DISCUSSION THREAD

**Discussion by ashleytarring** [nuke-build/nuke#1466] [Q&A]
Are there any pearls of wisdom, suggestions or documentation on Unit testing a pipeline or target once created?

Anyone have any examples of creating tests to execute a target? What would you assert on within your test.

Also a big problem i am having, how would you pass command line parameters via your tests? 
For example i need to pass "--profile unit" so that i can utilize a paramaters.unit.json file but how would you do this within test code, in ide context like visual studio, rather than from console command

**Comment by matkoch** [comment]
This has been brought up a couple of times, and so far I've always suggested to make a diff comparison on the build log (note that one log has timestamps removed). There are a couple of folks who seem to be interested in more fine-grained control, like mocking, parallel test execution, etc... I'm open to having some kind of discussion about that, but rather in verbal/voice format to explore it live and together. /cc @mmisztal1980 @lg2de
